core/xml.js

```diff
@@ -691,13 +728,44 @@ Blockly.Xml.domToBlockHeadless_ = function(xmlBlock, workspace) {
                           'Shadow block not allowed non-shadow child.');
     }
     // Ensure this block doesn't have any variable inputs.
-    goog.asserts.assert(block.getVars().length == 0,
-        'Shadow blocks cannot have variable fields.');
+    goog.asserts.assert(block.getVarModels().length == 0,
+        'Shadow blocks cannot have variable references.');
     block.setShadow(true);
   }
   return block;
 };
 
+/**
+ * Decode an XML variable field tag and set the value of that field.
+ * @param {!Blockly.Workspace} workspace The workspace that is currently being
+ *     deserialized.
+ * @param {!Element} xml The field tag to decode.
+ * @param {string} text The text content of the XML tag.
+ * @param {!Blockly.FieldVariable} field The field on which the value will be
+ *     set.
+ * @private
+ */
+Blockly.Xml.domToFieldVariable_ = function(workspace, xml, text, field) {
+  var type = xml.getAttribute('variabletype') || '';
+  // TODO (fenichel): Does this need to be explicit or not?
```

Comment:
is this still a TODO?

Reply:
Yes!  I can't convince myself one way or another.  I *think* if I don't make it explicit this way, it just doesn't get serialized if it's empty.